### ISSUE TITLE
Table accessible issue for HTML5 format - table cell headers information is not generated correctly

### ISSUE THREAD

**Issue by radu-pisoi** [dita-ot/dita-ot#2963]
The ``headers`` attribute (identifies the table headers associated with the cell) is not generated correctly for a table with header cells in the first column. 

A sample table:
```
<table rowheader="firstcol" >
        <title>Sample table</title>
        <tgroup cols="2">
          <colspec colname="c1" colnum="1" colwidth="1*"/>
          <colspec colname="c2" colnum="2" colwidth="1*"/>
          <tbody>
            <row>
              <entry>Event</entry>
              <entry>Evolution of TC 2018</entry>
            </row>
            <row>
              <entry>Date</entry>
              <entry>May 31 - June 1, 2018</entry>
            </row>
          </tbody>
        </tgroup>
</table>
```
The generated HTML5 table is:
```
<table class="table">
  <caption>
    <span class="table--title-label">Table 1. </span>
    <span class="title">Sample table</span>
  </caption>
  <colgroup>
    <col style="width:50%"/>
    <col style="width:50%"/>
  </colgroup>
  <tbody class="tbody">
    <tr class="row">
      <th scope="row" class="entry">Event</th>
      <td class="entry" headers="d379e32 ">Evolution of TC 2018</td>
    </tr>
    <tr class="row">
      <th scope="row" class="entry">Date</th>
      <td class="entry" headers="d379e41 ">May 31 - June 1, 2018</td>
    </tr>
  </tbody>
</table>
```
As you can see, there is no id attribute generated for the ``th`` elements.
Please note that this issue cannot be reproduced with the XHTML transformation.

More over, there are issues when DITA table cell specifies explicitly the cell headers information.
Sample table:
```
<table rowheader="firstcol" >
        <title>Sample table with headers specified</title>
        <tgroup cols="2">
          <colspec colname="c1" colnum="1" colwidth="1*"/>
          <colspec colname="c2" colnum="2" colwidth="1*"/>
          <tbody>
            <row>
              <entry id="h1">Event</entry>
              <entry headers="h1">Evolution of TC 2018</entry>
            </row>
            <row>
              <entry id="h2">Date</entry>
              <entry headers="h2">May 31 - June 1, 2018</entry>
            </row>
          </tbody>
        </tgroup>
</table>
```
The generated HTML5 table is:
```
<table class="table">
  <caption>
    <span class="table--title-label">Table 2. </span>
    <span class="title">Sample table with headers specified</span>
  </caption>
  <colgroup>
    <col style="width:50%"/>
    <col style="width:50%"/>
  </colgroup>
  <tbody class="tbody">
    <tr class="row">
      <th scope="row" class="entry" id="table_issue__h1">Event</th>
      <td class="entry" headers="d379e66 ">Evolution of TC 2018</td>
    </tr>
    <tr class="row">
      <th scope="row" class="entry" id="table_issue__h2">Date</th>
      <td class="entry" headers="d379e75 ">May 31 - June 1, 2018</td>
    </tr>
  </tbody>
</table>
```

This issue is related with [Success Criterion 1.3.1 - Info and Relationships](https://www.w3.org/WAI/GL/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html) (Level A), see [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H43.html) technique.

* DITA-OT version: _DITA-OT 3.0.4_
* Transformation type:  _HTML5_

**Comment by robander** [comment]
When I test the first issue reported here (no ID added for headers in the first row), it is resolved with the latest updates including the table fix from #4776. That said, I don't think that new PR should have fixed the issue, it looks like it was resolved earlier. With the first sample table above pasted into a DITA document, and built to HTML5, I get the following table from DITA-OT 4.4. Note that the id attributes are added to each header, and the `headers` attributes line up correctly:
```
<table class="table"><caption><span class="table--title-label">Table 1. </span><span class="title">Sample table</span></caption><colgroup><col style="width:50%"><col style="width:50%"></colgroup><tbody class="tbody">
                    <tr class="row">
                        <th scope="row" class="entry" id="topic_rvk_5rs_vjc__entry__1">Event</th>
                        <td class="entry" headers="topic_rvk_5rs_vjc__entry__1">Evolution of TC 2018</td>
                    </tr>
                    <tr class="row">
                        <th scope="row" class="entry" id="topic_rvk_5rs_vjc__entry__3">Date</th>
                        <td class="entry" headers="topic_rvk_5rs_vjc__entry__3">May 31 - June 1, 2018</td>
                    </tr>
</tbody></table>
```

**Comment by robander** [comment]
When I test the second table, the results are better than when the original issue was opened in 2018, because the `@id` and `@headers` attributes from the DITA source table are totally ignored, and replaced with automated ones that are correct. Automation is a much better way to manage accessibility than by manually using id/headers. So - it's not working like it's supposed to, but it's ensuring correct output based on the other accessibility markup.

This does raise a problem though, that when the ID attributes are replaced on the table header cell, any DITA content cross-referencing that cell will have a broken link. This seems extremely unlikely, but possible. I think a better fix would be to drop an `<a>` anchor at the start of the entry to preserve that ID, and let automation continue to handle the accessibility.

**Comment by robander** [comment]
Scratch that last comment - I was wrong, the second table is also working correctly for me in the latest OT. If I have an ID on the table entry, it is preserved using the normal syntax of `topicid__id`, and the headers attribute is updated to use the matching value. So I think this issue is fully resolved.

**Comment by robander** [comment]
Marking closed - the reported issues are both working for me now using DITA-OT 4.4 along with the fix #4776 that is coming in 4.4.1.
